packages/tx-construction/src/tx-builder/OutputBuilder.ts

```diff
@@ -132,6 +134,14 @@ export class TxOutputBuilder implements OutputBuilder {
       throw outputValidation;
     }
 
+    if (this.#partialOutput.handle) {
+      const resolution = await this.#handleProvider.resolveHandles({ handles: [this.#partialOutput.handle] });
+
+      if (resolution[0] !== null) {
+        txOut.handle = resolution[0];
```

Comment:
I wasn't sure about this at the time and I wanted to ask today about it. The type of `resolvedAdresses.cardano` is:

```export type CardanoAddress =
  | Cardano.ByronAddress
  | Cardano.PaymentAddress
  | Cardano.EnterpriseAddress
  | Cardano.PointerAddress;
```

But the type for `txOut.address` is `export type PaymentAddress = OpaqueString<'PaymentAddress'>;`.
This breaks the typings since some of the `CardanoAdress` types are not strings, but classes with additional properties. 

Do we need to try and check the type and attempt to resolve it to a string?  Or is the type of `HandleResolution.resolvedAddresses.cardano` actually just `PaymentAddress`?

Reply:
> Or is the type of HandleResolution.resolvedAddresses.cardano actually just PaymentAddress

That's correct, let's update the type.